org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceExecuteCommandHandler.java

```diff
@@ -119,24 +123,27 @@ public synchronized IDelegateCommandHandler getDelegateCommandHandler() {
 			}
 			return fDelegateCommandHandlerInstance;
 		}
+
+		public static String createId(IConfigurationElement element) {
+			return element.getNamespaceIdentifier() + "#" + element.getAttribute(CLASS);
+		}
 	}
 
-	private Set<DelegateCommandHandlerDescriptor> fgContributedCommandHandlers;
+	private final Map<String, DelegateCommandHandlerDescriptor> fgContributedCommandHandlers;
 
 	private WorkspaceExecuteCommandHandler() {
+		IConfigurationElement[] elements = Platform.getExtensionRegistry().getConfigurationElementsFor(EXTENSION_POINT_ID);
+		fgContributedCommandHandlers = Stream.of(elements).collect(Collectors.toMap(DelegateCommandHandlerDescriptor::createId, DelegateCommandHandlerDescriptor::new));

```

Comment:
it's more safe to add a merge function for toMap. `Collectors.toMap(DelegateCommandHandlerDescriptor::createId, DelegateCommandHandlerDescriptor::new, (value1, value2) -> value2)`

Reply:
Done, also on added method.